VMware.CloudFoundation.PasswordManagement.psm1

```diff
@@ -600,7 +600,7 @@ Function Get-PasswordPolicyDefault {
     Param (
         [Parameter (Mandatory = $false, ParameterSetName = 'json')] [ValidateNotNullOrEmpty()] [Switch]$generateJson,
         [Parameter (Mandatory = $true)] [ValidateSet('4.4.0','4.5.1','5.0.0')] [String]$version,
-        [Parameter (Mandatory = $false, ParameterSetName = 'json')] [ValidateNotNullOrEmpty()] [String]$jsonFile
+        [Parameter (Mandatory = $true, ParameterSetName = 'json')] [ValidateNotNullOrEmpty()] [String]$jsonFile
```

Comment:
What about also checking the path exists and a file is not already present ?

Reply:
C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile M:\
Out-File : Cannot find drive. A drive with the name 'M' does not exist.
At C:\GitHubPublic\VMware.CloudFoundation.PowerManagement\VMware.CloudFoundation.PasswordManagement.psm1:809 char:43
+ ...        $defaultConfig | ConvertTo-Json | Out-File -FilePath $jsonFile
+                                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (M:String) [Out-File], DriveNotFoundException
    + FullyQualifiedErrorId : DriveNotFound,Microsoft.PowerShell.Commands.OutFileCommand

Generated JSON File (M:\) with Product Password Policy Default Values
C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\Users\Administrator\passwordPolicyConfig.json
Generated JSON File (C:\Users\Administrator\passwordPolicyConfig.json) with Product Password Policy Default Values

Reply:
If path doesn't exists, it clearly states, but if the file is already present, it is overriding now, you want me to throw error if already present and stop execution immediately?

Reply:
The above could likely be put in a `do-while` loop so that it would continue prompting for a **valid** choice.

Reply:
C:\Users\Administrator> Import-Module 'C:\GitHubPublic\VMware.CloudFoundation.PowerManagement\VMware.CloudFoundation.PasswordManagement.psd1' -force
>> Import-Module 'C:\GitHubPublic\PowerValidatedSolutions\PowerValidatedSolutions.psd1' -force
C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\Users\Administrator
Get-PasswordPolicyDefault : The 'jsonfile' parameter having the value 'C:\Users\Administrator' contains folder name and not the filename. Please recheck.
At line:1 char:1
+ Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\U ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-PasswordPolicyDefault

C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\Users\Administrator\as
Get-PasswordPolicyDefault : The filename doesn't contain '.json' extension, please check.
At line:1 char:1
+ Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\U ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-PasswordPolicyDefault

C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\Users\Administrator\as.json
Generated JSON File (C:\Users\Administrator\as.json) with Product Password Policy Default Values
C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\Users\Administrator\as.json
Get-PasswordPolicyDefault : The filename C:\Users\Administrator\as.json already exists, unable to override. use -force parameter to override.
At line:1 char:1
+ Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\U ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Get-PasswordPolicyDefault

C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0 -jsonFile C:\Users\Administrator\as.json -force
WARNING: The filename C:\Users\Administrator\as.json already exists, overriding the same.
Generated JSON File (C:\Users\Administrator\as.json) with Product Password Policy Default Values
C:\Users\Administrator> Get-PasswordPolicyDefault -generateJson -version 5.0.0

cmdlet Get-PasswordPolicyDefault at command pipeline position 1
Supply values for the following parameters:
jsonFile:
C:\Users\Administrator> Get-PasswordPolicyDefault -version 5.0.0


vcf                : @{vcfVersion=5.0.0}
esxi               : @{passwordExpiration=; passwordComplexity=; accountLockout=}
sso                : @{passwordExpiration=; passwordComplexity=; accountLockout=}
vcenterServer      : @{passwordExpiration=}
vcenterServerLocal : @{passwordExpiration=; passwordComplexity=; accountLockout=}
nsxManager         : @{passwordExpiration=; passwordComplexity=; accountLockout=}
nsxEdge            : @{passwordExpiration=; passwordComplexity=; accountLockout=}
sddcManager        : @{passwordExpiration=; passwordComplexity=; accountLockout=}
wsaLocal           : @{passwordExpiration=; passwordComplexity=; accountLockout=}
wsaDirectory       : @{passwordExpiration=; passwordComplexity=; accountLockout=}